core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java

```diff
@@ -245,23 +244,17 @@ private InstanceData performRefresh() throws InterruptedException, ExecutionExce
     forcedRenewRateLimiter.acquirePermit();
 
     ListenableFuture<InstanceData> refreshFuture;
-    if (authType == AuthType.IAM) {
-      if (credentials.isPresent()) {
-        GoogleCredentials downscopedCredentials = getDownscopedCredentials(credentials.get());
-        refreshFuture =
-            apiFetcher.getInstanceData(
-                instanceName, downscopedCredentials, AuthType.IAM, executor, keyPair);
-      } else {
-        throw new RuntimeException(
```

Comment:
We probably still need to throw an exception when no credentials have been provided.

Reply:
I think that case is covered on line 104. There is no way to create a CloudSQLInstance with authType == IAM leaving this.iamAuthnCredentials with an empty value. Optional.of(null) will throw an exception.

Reply:
Ah yes. In that case, let's update the exception message above to match what we have here (which is more useful, I think).

Reply:
Fixed.